pkg/asset/installconfig/aws/validation.go

```diff
@@ -226,6 +226,9 @@ func validateSubnets(ctx context.Context, meta *Metadata, fldPath *field.Path, s
 	if err != nil {
 		return append(allErrs, field.Invalid(fldPath, subnets, err.Error()))
 	}
+	if publish == types.InternalPublishingStrategy && len(publicSubnets) > 0 {
+		logrus.Warnf("Public subnets can not be set when publish is set to %s", types.InternalPublishingStrategy)
```

Comment:
What about:
```suggestion
		logrus.Warnf("Public subnets should not be provided when publish is set to %s", types.InternalPublishingStrategy)
```
The "can not" implies this should be an error condition but we agreed on just displaying a warning.